### DISCUSSION TITLE
command / cmd exec error handling

### DISCUSSION THREAD

**Discussion by sjwl** [cue-lang/cue#2415] [Q&A] [answered]
How can I catch errors from `exec.Run` commands and gracefully handle it?

Given the example below, if I run `cue demo .` instead of printing out "failed", it prints `task failed: command "ls does-not-exist" failed: exit status 1`
```cue
package foo

import (
        "tool/cli"
        "tool/exec"
)

command: demo: {
        apply: exec.Run & {
                cmd: ["ls", "does-not-exist"]
                stdout: string
                stderr: string
        }

        result: [
                if apply.success {
                        "success"
                },
                if !apply.success {
                        "failed"
                },
        ]

        print: cli.Print & {
                text: result[0]
        }
}
```

**Comment by verdverm** [comment, marked as answer]
Open issue and work around: https://github.com/cue-lang/cue/issues/1499

**Reply by sjwl** [reply]
thanks for the response.  The workaround seems to be forking my own version of `tool/exec` pkg?
